### DISCUSSION TITLE
rewrites will render twice

### DISCUSSION THREAD

**Discussion by haoflynet** [vercel/next.js#27985] [Help] [answered]
I'm trying the [example project](https://github.com/mui-org/material-ui/tree/master/examples/nextjs) of `material-ui`, and if I add this code to my `next.config.js`, the page will render twice. I just visit `/about`, not `/`, it still render twice (check by adding a console.log in `about.js`), and if I remove the `rewites` in `next.config.js`, it will just render once like normal :

```
module.exports = {
    async rewrites() {
        return [
            {
                source: '/',
                destination: '/about',
            },
        ]
    }
}

```

**Comment by elyash14** [comment]
I have the same issue, In my project, I am using the rewrites feature to support multiple domains. But every page which uses rewights is rendered twice.

**Comment by ijjk** [comment, marked as answer]
Hi, this is the expected behavior and a note was added in the documentation for this here https://github.com/vercel/next.js/pull/30747
